### PR TITLE
SBERDOMA-308 Sms adapter

### PR DESCRIPTION
**Pull request by dkoviazin**
1. smsAdapter with 2 providers sms.ru or smsc.ru
2. add template and message type for phone verification code
3. add tests. tests will not fail on misconfigured environment

Maybe need to move adapters from common domain to notifications domain

### CODE REVIEW

File: apps/condo/domains/common/utils/smsAdapter.js [hunk 1]
```diff
@@ -0,0 +1,216 @@
+const { isEmpty, get, has } = require('lodash')
```
Comment (pahaz) [inline apps/condo/domains/common/utils/smsAdapter.js:1]: please move it inside notifications utils

Comment (dkoviazin) [reply apps/condo/domains/common/utils/smsAdapter.js:1]: Ok

File: apps/condo/domains/common/utils/smsAdapter.js [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+    get isConfigured () {
+        return this.adapter.isConfigured
+    }
+
+    async checkIsAvailable () {
+        const result = await this.adapter.checkIsAvailable()
+        return result
+    }
+
+    async sendTestSMS ({ phone, message }) {
```
Comment (Dimitreee) [inline apps/condo/domains/common/utils/smsAdapter.js:43]: what is this class method for?

File: apps/condo/domains/common/utils/smsAdapter.js [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+}
+
+
+class SmsRu {
+    
+    isConfigured = false
+
+    constructor () {
+        //TODO(zuch): Remove old env record
+        const config = conf['SMSRU_API_CONFIG'] ?  JSON.parse(conf['SMSRU_API_CONFIG']) :
```
Comment (Dimitreee) [inline apps/condo/domains/common/utils/smsAdapter.js:72]: Avoid nested ternary operators, constructions like this make your code less readable than simplified `if else`.

File: apps/condo/domains/common/utils/smsAdapter.js [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    constructor () {
+        //TODO(zuch): Remove old env record
+        const config = conf['SMSRU_API_CONFIG'] ?  JSON.parse(conf['SMSRU_API_CONFIG']) : 
+            conf['SMS_API_CONFIG'] ? JSON.parse(conf['SMS_API_CONFIG']) : 
+                {}
+        this.isConfigured = validateConfig(config, [
+            'token',
+            'from',
+        ])
+        this.api_url = 'https://sms.ru'
```
Comment (Dimitreee) [inline apps/condo/domains/common/utils/smsAdapter.js:79]: move to conf, please

File: apps/condo/domains/common/utils/smsAdapter.js [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+        this.token = config.token
+        this.from = config.from
+    }
+
+    isPhoneSupported (phoneNumber) {
+        return /^[+]7[0-9]{10}$/g.test(phoneNumber)
+    }
+
+    async checkIsAvailable () {
+        const result = await fetch(
```
Comment (Dimitreee) [inline apps/condo/domains/common/utils/smsAdapter.js:89]: does fetch works well in node.js env?

File: apps/condo/domains/common/utils/smsAdapter.js [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+            `${this.api_url}/auth/check`,
+            {
+                method: 'POST',
+                headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
+                body: `api_id=${this.token}&json=1`,
+            }
+        )
+        const json = await result.json()
+        const status = json['status_code']
+        const isOk = status === 100
```
Comment (Dimitreee) [inline apps/condo/domains/common/utils/smsAdapter.js:99]: 200 maybe?

File: apps/condo/domains/common/utils/smsAdapter.js [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+            json: 1,
+            msg: message,
+            ...extendedParams,
+        }
+        const result = await fetch(
+            `${this.api_url}/sms/send`,
+            {
+                method: 'POST',
+                headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
+                body:  Object.entries(body)
```
Comment (Dimitreee) [inline apps/condo/domains/common/utils/smsAdapter.js:122]:
```suggestion
                body: Object.entries(body)
```

File: apps/condo/domains/common/utils/smsAdapter.js [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+    isConfigured = false
+
+    constructor () {
+        const config = conf['SMSCRU_API_CONFIG'] ? JSON.parse(conf['SMSCRU_API_CONFIG']) : {}
+        this.isConfigured = validateConfig(config, [
+            'login',
+            'password',
+            'sender',
+        ])
+        this.api_url = 'https://smsс.ru'
```
Comment (Dimitreee) [inline apps/condo/domains/common/utils/smsAdapter.js:145]: move to conf

**Review by Dimitreee — APPROVED** [review]

**Review by AntonAL — APPROVED** [review]

File: apps/condo/domains/common/utils/smsAdapter.js (hunk 1 above)
Comment (dkoviazin) [reply apps/condo/domains/common/utils/smsAdapter.js:1]: Done

File: apps/condo/domains/common/utils/smsAdapter.js (hunk 2 above)
Comment (dkoviazin) [reply apps/condo/domains/common/utils/smsAdapter.js:43]: removed

File: apps/condo/domains/common/utils/smsAdapter.js (hunk 3 above)
Comment (dkoviazin) [reply apps/condo/domains/common/utils/smsAdapter.js:72]: Fixed

File: apps/condo/domains/common/utils/smsAdapter.js (hunk 4 above)
Comment (dkoviazin) [reply apps/condo/domains/common/utils/smsAdapter.js:79]: Done

File: apps/condo/domains/common/utils/smsAdapter.js (hunk 5 above)
Comment (dkoviazin) [reply apps/condo/domains/common/utils/smsAdapter.js:89]: it's node fetch
```js
const fetch = require('node-fetch')
```

File: apps/condo/domains/common/utils/smsAdapter.js (hunk 6 above)
Comment (dkoviazin) [reply apps/condo/domains/common/utils/smsAdapter.js:99]: I's their custom code
```js
{
    "status": "OK", // Запрос выполнен успешно (нет ошибок в авторизации, проблем с отправителем, итд...)
    "status_code": 100, // Успешный код выполнения
    "sms": {
        "79255070602": {
            "status": "OK", // Возможные варианты: OK или ERROR.
            "status_code": 100, // Успешный код выполнения, сообщение принято на отправку
            "sms_id": "000000-10000000" // ID сообщения
        },
        "74993221627": {
            "status": "ERROR",
            "status_code": 207, // Код ошибки
            "status_text": "На этот номер (или один из номеров) нельзя отправлять сообщения, либо указано более 100 номеров в списке получателей" // Описание ошибки
        }
    } 
    "balance": 4122.56 // Ваш баланс после отправки
}
```

File: apps/condo/domains/common/utils/smsAdapter.js (hunk 7 above)
Comment (dkoviazin) [reply apps/condo/domains/common/utils/smsAdapter.js:122]: Done

File: apps/condo/domains/common/utils/smsAdapter.js (hunk 8 above)
Comment (dkoviazin) [reply apps/condo/domains/common/utils/smsAdapter.js:145]: done

File: apps/condo/domains/notification/templates.js [hunk 9]
```diff
@@ -67,16 +73,30 @@ async function renderTemplate (transport, message) {
             }
         } else if (message.lang === 'ru') {
             return {
-                subject: '🔑 Восстановление пароля',
```
Comment (pahaz) [inline apps/condo/domains/notification/templates.js:70]: we don't need an emoji ? :(

**Review by pahaz — APPROVED** [review]

File: apps/condo/domains/notification/templates.js (hunk 9 above)
Comment (dkoviazin) [reply apps/condo/domains/notification/templates.js:70]: I put it here and then realized we needed more consistency - do we use them everywhere or not at all :)

**Pull request merged**